ndarray/src/main/java/org/tensorflow/ndarray/impl/sparse/SparseNdArray.java

```diff
@@ -394,4 +394,25 @@ public U toDense() {
   public Class<T> getType() {
     return type;
   }
+
+  /**
+   * A String showing the type, default value, number of elements and
+   * the dense shape of this sparse ndarray.
+   * @return A string containing the type, default value, number of elements and shape.
+   */
+  @Override
+  public String toString() {
+    long numElements = getValues() == null ? 0 : getValues().size();
+    String strDefault;
+    T defaultVal = getDefaultValue();
+    if (defaultVal == null) {
+      strDefault = "<null>";
+    } else if (defaultVal instanceof Number){
+      strDefault = defaultVal.toString();
+    } else {
+      strDefault = "'" + defaultVal + "'";
+    }
+    return this.getClass().getSimpleName() + "(type="+type.getSimpleName()+",defaultValue=" + strDefault
+            + ",numElements=" + numElements + ",shape=" + this.shape() + ")";
```

Comment:
Do we still need this since it is implemented by the AbstractSparseNdArray the same way, which is extended by this class?

Reply:
It's slightly different as the abstract one doesn't print the type (because it's assumed that it'll be in the name, and it's not accessible from `AbstractSparseNdArray`) and the one in `SparseNdArray` has access to a type field.

Reply:
aaaah ok got it, sounds good, I've missed that